repositories.bzl

```diff
@@ -0,0 +1,559 @@
+        "include/tclap/CmdLineInterface.h",
+        "include/tclap/CmdLineOutput.h",
+        "include/tclap/Constraint.h",
+        "include/tclap/DocBookOutput.h",
+        "include/tclap/HelpVisitor.h",
+        "include/tclap/IgnoreRestVisitor.h",
+        "include/tclap/MultiArg.h",
+        "include/tclap/MultiSwitchArg.h",
+        "include/tclap/OptionalUnlabeledTracker.h",
+        "include/tclap/StandardTraits.h",
+        "include/tclap/StdOutput.h",
+        "include/tclap/SwitchArg.h",
+        "include/tclap/UnlabeledMultiArg.h",
+        "include/tclap/UnlabeledValueArg.h",
+        "include/tclap/ValueArg.h",
+        "include/tclap/ValuesConstraint.h",
+        "include/tclap/VersionVisitor.h",
+        "include/tclap/Visitor.h",
+        "include/tclap/XorHandler.h",
+        "include/tclap/ZshCompletionOutput.h",
+    ],
+    defines = [
+        "HAVE_LONG_LONG=1",
+        "HAVE_SSTREAM=1",
+    ],
+    includes = [
+        "include",
+    ],
+    visibility = ["//visibility:public"],
+)"""
+
+    native.new_http_archive(
+        name = "tclap_tar",
+        url = "https://storage.googleapis.com/istio-build-deps/tclap-1.2.1.tar.gz",
+        strip_prefix = "tclap-1.2.1",
+        build_file_content = BUILD,
+    )
+    if bind:
+        native.bind(
+            name = "tclap",
+            actual = "@tclap_tar//:tclap",
+        )
+
+def lightstep_repositories(bind=True):
+    BUILD = """
+load("@protobuf_git//:protobuf.bzl", "cc_proto_library")
+
+genrule(
+    name = "envoy_carrier_pb",
+    srcs = ["src/c++11/envoy/envoy_carrier.proto"],
+    outs = ["lightstep/envoy_carrier.proto"],
+    cmd = "cp $(SRCS) $@",
+)
+
+cc_proto_library(
+    name = "envoy_carrier_proto",
+    srcs = ["lightstep/envoy_carrier.proto"],
+    include = ".",
+    deps = [
+        "//external:cc_wkt_protos",
+    ],
+    protoc = "//external:protoc",
+    default_runtime = "//external:protobuf",
+    visibility = ["//visibility:public"],
+)
+
+cc_library(
+    name = "lightstep_core",
+    srcs = [
+        "src/c++11/impl.cc",
+        "src/c++11/span.cc",
+        "src/c++11/tracer.cc",
+        "src/c++11/util.cc",
+    ],
+    hdrs = [
+        "src/c++11/lightstep/impl.h",
+        "src/c++11/lightstep/options.h",
+        "src/c++11/lightstep/propagation.h",
+        "src/c++11/lightstep/envoy.h",
+        "src/c++11/lightstep/span.h",
+        "src/c++11/lightstep/tracer.h",
+        "src/c++11/lightstep/util.h",
+        "src/c++11/lightstep/value.h",
+        "src/c++11/mapbox_variant/recursive_wrapper.hpp",
+        "src/c++11/mapbox_variant/variant.hpp",
+    ],
+    copts = [
+        "-DPACKAGE_VERSION='\\"0.19\\"'",
+        "-Iexternal/lightstep_git/src/c++11/lightstep",
+        "-Iexternal/lightstep_git/src/c++11/mapbox_variant",
+    ],
+    includes = [
+        "src/c++11",
+    ],
+    visibility = ["//visibility:public"],
+    deps = [
+        "@lightstep_common_git//:collector_proto",
```

Comment:
Buildifier: `@lightstep_common_git//:collector_proto` should be listed at the end (alphabetical order).